### PR TITLE
Add unit tests for ContextMenuStripActionListTests

### PR DESCRIPTION
**Pull request by MelonWang1**
Related https://github.com/dotnet/winforms/issues/10773
## Proposed changes

- Add unit test ContextMenuStripActionListTests.cs for public properties and method of the ContextMenuStripActionList.
- Enable nullability in ContextMenuStripActionList.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12442)

### CODE REVIEW

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ContextMenuStripActionListTests.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.ComponentModel.Design;
+using System.ComponentModel;
+using Moq;
+
+namespace System.Windows.Forms.Design.Tests;
+public sealed class ContextMenuStripActionListTests : IDisposable
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ContextMenuStripActionListTests.cs:11]: nit:
```suggestion
namespace System.Windows.Forms.Design.Tests;

public sealed class ContextMenuStripActionListTests : IDisposable
```

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ContextMenuStripActionListTests.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        _actionList.ShowCheckMargin.Should().BeTrue();
+
+        _actionList.ShowCheckMargin = false;
+        _actionList.ShowCheckMargin.Should().BeFalse();
+    }
+
+    [Fact]
+    public void RenderMode_GetSet_ReturnsExpected()
+    {
+        _actionList.RenderMode = ToolStripRenderMode.System;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ContextMenuStripActionListTests.cs:95]: Could you please test ass 4 RenderModes - https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.toolstriprendermode?view=windowsdesktop-8.0. This test is a good candidate for a conversion to a Theory

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ContextMenuStripActionListTests.cs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        _actionList.AutoShow.Should().BeFalse();
+
+        _actionList.AutoShow = true;
+        _actionList.AutoShow.Should().BeTrue();
+    }
+
+    [Fact]
+    public void ShowImageMargin_GetSet_ReturnsExpected()
+    {
+        _actionList.ShowImageMargin = true;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ContextMenuStripActionListTests.cs:75]: Could you please add a test for the default value? Also consider changing this one to a theory, you can use [BoolData] as the input.

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ContextMenuStripActionListTests.cs:75]: The same comment applies to the next test.

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ContextMenuStripActionListTests.cs [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+        items.Should().Contain(item => item.MemberName == "RenderMode");
+        items.Should().Contain(item => item.DisplayName == SR.ToolStripActionList_RenderMode);
+        items.Should().Contain(item => item.Description == SR.ToolStripActionList_RenderModeDesc);
+    }
+
+    [Fact]
+    public void GetSortedActionItems_WithToolStripDropDownMenu_IncludesShowImageMarginAndShowCheckMargin()
+    {
+        var items = _actionList.GetSortedActionItems().Cast<DesignerActionPropertyItem>().ToList();
```
Comment (ricardobossan) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ContextMenuStripActionListTests.cs:115]: This method body could be more concise and readable by chaining assertions like this:

```cs
var items = _actionList.GetSortedActionItems().Cast<DesignerActionPropertyItem>().ToList();

items.Select(item => item.MemberName)
    .Should().Contain("RenderMode")
    .And.Contain("ShowImageMargin")
    .And.Contain("ShowCheckMargin");

items.Select(item => item.DisplayName)
    .Should().Contain(SR.ContextMenuStripActionList_ShowImageMargin)
    .And.Contain(SR.ContextMenuStripActionList_ShowCheckMargin);
```

Comment (ricardobossan) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ContextMenuStripActionListTests.cs:115]: This method body could be more concise and readable by chaining assertions like this:

```cs
var items = _actionList.GetSortedActionItems().Cast<DesignerActionPropertyItem>().ToList();

items.Select(item => item.MemberName)
    .Should().Contain("RenderMode")
    .And.Contain("ShowImageMargin")
    .And.Contain("ShowCheckMargin");

items.Select(item => item.DisplayName)
    .Should().Contain(SR.ContextMenuStripActionList_ShowImageMargin)
    .And.Contain(SR.ContextMenuStripActionList_ShowCheckMargin);
```

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ContextMenuStripActionListTests.cs [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        _actionList.RenderMode = ToolStripRenderMode.Professional;
+        _actionList.RenderMode.Should().Be(ToolStripRenderMode.Professional);
+    }
+
+    [Fact]
+    public void GetSortedActionItems_AlwaysReturnsRenderMode()
+    {
+        var items = _actionList.GetSortedActionItems().Cast<DesignerActionPropertyItem>().ToList();
+
+        items.Should().Contain(item => item.MemberName == "RenderMode");
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ContextMenuStripActionListTests.cs:107]: Please use `nameof(ContextMenuStripActionList.RenderMode)` to be explicit where this text is coming from.
The same comment applies to the next 2 tests.

**Review by Tanya-Solyanik — COMMENTED** [review]
Looks good!

**Review by ricardobossan — COMMENTED** [review]
Other than small comments, LGTM.

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ContextMenuStripActionListTests.cs [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+        _actionList.ShowCheckMargin = value;
+        _actionList.ShowCheckMargin.Should().Be(value);
+    }
+
+    [Theory]
+    [InlineData(ToolStripRenderMode.System)]
+    [InlineData(ToolStripRenderMode.Professional)]
+    [InlineData(ToolStripRenderMode.ManagerRenderMode)]
+    [InlineData(ToolStripRenderMode.Custom)]
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ContextMenuStripActionListTests.cs:98]: If this is all of the values for `ToolStripRenderMode` consider changing this to 
```suggestion
    [EnumData<ToolStripRenderMode>]
```

**Review by lonitra — COMMENTED** [review]
LGTM 👍

**Review by lonitra — APPROVED** [review]

**Pull request merged**